### PR TITLE
Add setup_gnome_display_manager playbook

### PR DESCRIPTION
**Pull request by HVSharma12**
Add setup_gnome_display_manager playbook

### CODE REVIEW

File: examples/ansible/setup_gnome_display_manager.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  hosts: alphost
+  become: true
+  vars:
+    workload:
+      name: gdm
+      image: registry.opensuse.org/suse/alp/workloads/tumbleweed_containerfiles/suse/alp/workloads/gdm:latest
+      required_pkgs:
+        - accountsservice
+        - systemd-experimental
+        - python3-selinux
```
Comment (pdostal) [inline examples/ansible/setup_gnome_display_manager.yml]: Again, this is static. Why is it defined as variable?

File: examples/ansible/setup_gnome_display_manager.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  tasks:
+    - name: Install required packages, if any, for workload {{ workload.name }}
+      ansible.builtin.package:
+        name: "{{ workload.required_pkgs }}"
+        state: present
+      register: required_pkgs_install
+      when:
+        - ((workload.required_pkgs | default([])) | length) > 0
```
Comment (pdostal) [inline examples/ansible/setup_gnome_display_manager.yml]: What doesthis even mean?

File: examples/ansible/setup_gnome_display_manager.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      required_pkgs:
+        - accountsservice
+        - systemd-experimental
+        - python3-selinux
+
+  tasks:
+    - name: Install required packages, if any, for workload {{ workload.name }}
+      ansible.builtin.package:
+        name: "{{ workload.required_pkgs }}"
+        state: present
```
Comment (pdostal) [inline examples/ansible/setup_gnome_display_manager.yml]:
```suggestion
      ansible.builtin.package:
        name: ['accountsservice', 'systemd-experimental', 'python3-selinux']
        state: present
```

File: examples/ansible/setup_gnome_display_manager.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      register: required_pkgs_install
```
Comment (pdostal) [inline examples/ansible/setup_gnome_display_manager.yml]:
```suggestion
      notify: reboot
    - name: Reboot right now if necessary
      meta: flush_handlers
            
handlers:
  - name: reboot
    reboot:
      reboot_timeout: 600
      post_reboot_delay: 60
```

File: examples/ansible/setup_gnome_display_manager.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+              Required system packages have been installed via
+              transactional-update to support workload
+              {{ workload.name }}.
+              An immediate reboot is required to activate these
+              packages before proceding. Once reboot has completed
+              please re-run this playbook.
+
+        - name: Request immediate reboot  # noqa no-handler
+          ansible.builtin.fail:
+            msg: "Reboot required"
```
Comment (pdostal) [inline examples/ansible/setup_gnome_display_manager.yml]: This is not needed.

File: examples/ansible/setup_gnome_display_manager.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+    - name: Reload dbus service
+      ansible.builtin.systemd:
+        name: dbus
+        state: reloaded
+
+    - name: Restart accounts-daemon service
+      ansible.builtin.systemd:
+        name: accounts-daemon
+        state: restarted
```
Comment (pdostal) [inline examples/ansible/setup_gnome_display_manager.yml]: We just installed it. I suppose it only needs to be running?
```suggestion
        state: started
        enabled: true
```

File: examples/ansible/setup_gnome_display_manager.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    - name: Apply container runlabel install for workload {{ workload.name }}
+      ansible.builtin.command: >-
+        podman container runlabel install {{ workload.image }}
+      register: workload_runlabel_install
+      changed_when:
+        - ('already exist' not in workload_runlabel_install.stdout)
+
+    - name: Reload systemd daemon
+      ansible.builtin.systemd:
+        daemon_reload: true
```
Comment (pdostal) [inline examples/ansible/setup_gnome_display_manager.yml:45]: just nit
```suggestion
        daemon_reload: true
      when: workload_runlabel_install.stdout.changed
```

**Review by pdostal — CHANGES_REQUESTED** [review]
I feel that installing packages via Ansible on transactional system from that system itself is a bit nonsense as the system should be 'production' and the Ansible should be run from a client.

The Ansible can then reboot that target system and continue after that just fine.

**Review by pdostal — APPROVED** [review]
All good from my side. Just a question:

Before you merge such pull request, do you actually try to run the playbook somewhere?

**Comment by HVSharma12** [comment]
@pdostal Yes tested it on [ALP Dolomite 1.0 Milestone 3](https://susealp.io/downloads/alp-dolomite-1.0-milestone-3/)

**Comment by pdostal** [comment]
Very cool :smile:

**Review by brett060102 — APPROVED** [review]
Looks great.

**Pull request merged**